packages/mdc-textfield/mdc-textfield.scss

```diff
@@ -313,32 +305,88 @@ $mdc-textfield-disabled-border-on-dark: rgba(white, .3);
   // stylelint-enable plugin/selector-bem-pattern
 }
 
+.mdc-textfield--with-leading-icon .mdc-textfield__icon,
+.mdc-textfield--with-trailing-icon .mdc-textfield__icon {
+  position: absolute;
+  bottom: 16px;
+  cursor: pointer;
+
+  @include mdc-theme-dark(".mdc-textfield") {
+    @include mdc-theme-prop(color, text-secondary-on-dark);
+  }
+}
+
+.mdc-textfield--with-leading-icon {
+  .mdc-textfield__input {
+    @include mdc-rtl-reflexive-property(padding, $mdc-textfield-icon-padding, $mdc-textfield-icon-position);
+  }
+
+  .mdc-textfield__icon {
+    @include mdc-rtl-reflexive-position(left, $mdc-textfield-icon-position);
+  }
+
+  .mdc-textfield__label {
+    @include mdc-rtl-reflexive-position(left, $mdc-textfield-icon-padding);
```

Comment:
Is there intentionally not an analogue to this block below for trailing icons? (And ditto for dense styles below these)

Reply:
Yes. It doesn't need the extra leading padding because there is no icon preceding it.